src/test/CommandSetTest.cpp

```diff
@@ -0,0 +1,86 @@
+#include "CommandSet.hpp"
+
+#include <catch2/catch.hpp>
+
+#include <sstream>
+#include <string_view>
+
+template <typename T>
+std::string enumerate_str(const CommandSet<T> &cs, int min_level, int max_level) {
```

Comment:
Love it!
Would you mind wrapping it in an anonymous namespace (the "improved" `static`)... just reduces the chance of two files defining the same function by accident.

Reply:
good point. Plus, it doesn't destroy all debug stuff.

Reply:
> good point. Plus, it doesn't destroy all debug stuff.

Which what? :)

Reply:
Oh, was thinking of in real code: 'static' symbols tend to get ditched by the compiler, but this is a unit test, so not really important here.

Reply:
Right. `static` means the link-level symbols are not defined, yes. But (in theory) not lost from the debug symbols. I tend to build everything with debug symbols on (even release builds) for this reason!